druid-shell/src/platform/windows/window.rs

```diff
@@ -1369,6 +1378,89 @@ impl WindowHandle {
     // TODO: Implement this
     pub fn show_titlebar(&self, _show_titlebar: bool) {}
 
+    pub fn set_position(&self, position: Point) {
+        if let Some(w) = self.state.upgrade() {
+            let hwnd = w.hwnd.get();
+            unsafe {
+                let mut rect = RECT { left: 0, top: 0, right: 0, bottom: 0};
+                
+                if GetWindowRect(hwnd, &mut rect) == 0 {
+                    warn!(
+                        "failed to get window rect: {}",
+                        Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                    );
+                };
+                let width = rect.right - rect.left;
+                let height = rect.bottom - rect.top;
+                if MoveWindow(hwnd, position.x as i32, position.y as i32, width, height, 0) == 0 {
+                    warn!(
+                        "failed to reposition window: {}",
+                        Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                    );
+                };
+            }
+        }
+    }
+
+    pub fn get_position(&self) -> Point {
+        if let Some(w) = self.state.upgrade() {
+            let hwnd = w.hwnd.get();
+            unsafe {
+                let mut rect = RECT { left: 0, top: 0, right: 0, bottom: 0};
+                if GetWindowRect(hwnd, &mut rect) == 0 {
+                    warn!(
+                        "failed to get window rect: {}",
+                        Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                    );
+                };
+                return Point::new(rect.left as f64, rect.top as f64)
+            }
+        }
+        warn!(
+            "failed to get window hwnd"
+        );
+        Point::new(0.0, 0.0)
+    }
+
+    pub fn set_size(&self, size: Size) {
+        if let Some(w) = self.state.upgrade() {
+            let hwnd = w.hwnd.get();
+            unsafe {
+                let position = self.get_position();
+                if MoveWindow(hwnd, position.x as i32, position.y as i32, size.width as i32, size.height as i32, 1) == 0 {
+                    warn!(
+                        "failed to resize window: {}",
+                        Error::Hr(HRESULT_FROM_WIN32(GetLastError()))
+                    );
+                };
+                // The GUI needs to redraw / layout components again here
+                // Unsure why it does not do this, MoveWindow sends the same messages as when you grab corner of window.
+            }
+        }
+    }
```

Comment:
Anyone know how I could tell the GUI that the window size has changed here?
The GUI needs to layout and paint components again.

Reply:
Look into the `WM_SIZE` handling code.

Reply:
But shouldn't that run when it receives WM_SIZE? I also tried sending it this way:
let lparam = MAKELONG(size.width as u16, size.height as u16);
w.wndproc.window_proc(hwnd, WM_SIZE, 0, lparam as isize);

Reply:
Oh I see the problem, to change the size I need the hwnd and to get that I have to call self.state.upgrade() to borrow the state. When the window then receives WM_SIZE it needs to borrow the state as mutable self.state.try_borrow_mut(). But since I am already borrowing it this fails, and instead of redrawing the GUI it does nothing